sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -48,7 +46,60 @@
     gen_sdc_pandas_rolling_overload_body, sdc_pandas_rolling_docstring_tmpl)
 from sdc.datatypes.common_functions import TypeChecker
 from sdc.hiframes.pd_dataframe_ext import get_dataframe_data
-from sdc.utils import sdc_overload_method
+from sdc.utils import sdc_overload_method, sdc_overload_attribute
+
+
+@sdc_overload_attribute(DataFrameType, 'index')
+def hpat_pandas_dataframe_index(df):
+    """
+       Intel Scalable Dataframe Compiler User Guide
+       ********************************************
+       Pandas API: pandas.DataFrame.index
+       Examples
+       --------
+       .. literalinclude:: ../../../examples/dataframe_index.py
+          :language: python
+          :lines: 27-
+          :caption: The index (row labels) of the DataFrame.
+          :name: ex_dataframe_index
+
+       .. command-output:: python ./dataframe_index.py
+           :cwd: ../../../examples
+
+       Intel Scalable Dataframe Compiler Developer Guide
+       *************************************************
+       Pandas DataFrame attribute :attr:`pandas.DataFrame.index` implementation.
+       .. only:: developer
+       Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_index*
+       Parameters
+       -----------
+       df: :obj:`pandas.DataFrame`
+           input arg
+       Returns
+       -------
+       :obj: `numpy.array`
+           return the index of DataFrame
+    """
+
+    ty_checker = TypeChecker(f'Attribute index.')
+    ty_checker.check(df, DataFrameType)
+
+    if isinstance(df.index, types.NoneType) or df.index is None:
+        empty_df = not df.columns
+
+        def hpat_pandas_df_index_none_impl(df):
+            if empty_df == True:  # nq
```

Comment:
Sorry for misleading, you need to replace `# nq` with `# noqa`

Reply:
Don't you want to try it in a single line?
`df_len = 0 if empty_df == False else len(get_dataframe_data(df, 0))`

But you need to check that it works.